### PR TITLE
Add FAQ page

### PR DESCRIPTION
**Pull request by nmattia**
## Description

This adds an FAQ page that is linked to from
1. The navbar
2. Anywhere where relevant (like the "confirm request" page to clarify what "application-specific" pseudonym means)

## Screenshots (if appropriate):

https://user-images.githubusercontent.com/6930756/129713084-d19e82cd-4a20-49fe-9389-1d2c71fd1bae.mov

## Notes

Our test framework (jest) made some of this more complicated than intended. It fails to run tests on a file that imports `lit-html`. Two workaround had to be put in place: first, the "questions" are split into a separate file (we're testing that anchors are unique). Second, the links are not inlined in the answers (but this turns out to look quite nice).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/frontend/src/flows/faq/questions.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// limitation above we can't really inline the links, because that would mean
+// importing `lit-html` here...
+
+export const questions = {
+  windowsHello: {
+    question: "Does Internet Identity support Windows Hello?",
+    anchor: "windows-hello",
+    answer:
+      "Yes! Internet Identity supports authenticating via Windows Hello. If Windows Hello is set up on your PC then Internet Identity will offer you to authenticate through Windows Hello.",
+    links: [],
```
Comment (kritzcreek) [inline src/frontend/src/flows/faq/questions.ts:17]: Lets point people here: https://sdk.dfinity.org/docs/ic-identity-guide/hello-guide.html

File: src/frontend/src/flows/faq/index.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      border-radius: 0.5rem;
+      padding: 0.5rem;
+    }
+    :target {
+      animation-name: example;
+      animation-duration: 3s;
+    }
+  </style>
+  <div class="container" id="faq">
+    <h1>FAQ</h1>
```
Comment (kritzcreek) [inline src/frontend/src/flows/faq/index.ts:43]:
```suggestion
    <h1>Frequently asked questions</h1>
```

File: src/frontend/src/flows/faq/index.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      color: var(--grey-800);
+    }
+
+    ul {
+      list-style-type: none;
+    }
+
+    li {
+      border-radius: 0.5rem;
+      padding: 0.5rem;
```
Comment (kritzcreek) [inline src/frontend/src/flows/faq/index.ts:35]:
```suggestion
      padding: 0.5rem 0;
```
I don't think the border radius is doing anything here? Also we're already pretty narrow, so let's use the horizontal space we have to put text on the screen.

File: src/frontend/src/flows/faq/index.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import { questions } from "./questions";
+
+// re-export for ease of use
+export { questions } from "./questions";
+
+// The FAQ page
+const pageContent = html`
+  <style>
+    /* briefly flash the question when redirected to a particular question */
+    @keyframes example {
```
Comment (kritzcreek) [inline src/frontend/src/flows/faq/index.ts:13]: Let's call this animation something like `flash-question`?

File: src/frontend/src/flows/faq/index.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    :target {
+      animation-name: example;
+      animation-duration: 3s;
+    }
+  </style>
+  <div class="container" id="faq">
+    <h1>FAQ</h1>
+    <ul>
+      ${Object.values(questions).map(
+        (faq) =>
```
Comment (kritzcreek) [inline src/frontend/src/flows/faq/index.ts:46]: How'd you feel about extracting this function into a `renderQuestion`?

File: src/frontend/src/index.ts [hunk 6]
```diff
@@ -5,14 +5,22 @@ import { addDevice } from "./flows/addDevice";
 import { renderManage } from "./flows/manage";
 import { compatibilityNotice } from "./flows/compatibilityNotice";
 import { aboutView } from "./flows/about";
+import { faqView } from "./flows/faq";
 import { intentFromUrl } from "./utils/userIntent";
 import { hasRequiredFeatures } from "./utils/featureDetection";
 import { displaySingleDeviceWarning } from "./flows/displaySingleDeviceWarning";
 import { setupRecovery } from "./flows/recovery/setupRecovery";
 import { IIConnection } from "./utils/iiConnection";
+import { confirmRedirect } from "./flows/confirmRedirect";
```
Comment (kritzcreek) [inline src/frontend/src/index.ts:14]: unused

File: src/frontend/src/styles/main.css [hunk 7]
```diff
@@ -269,7 +269,7 @@ hr {
   margin-top: -3rem;
 }
 
-#aboutLink {
+#navbar a {
```
Comment (kritzcreek) [inline src/frontend/src/styles/main.css:272]: This way you're losing the `margin-top` on the navbar (You'll have to put that on `#navbar` directly). I'd also rather target the links with a class, avoiding nested selectors as much as possible.

File: src/frontend/src/flows/faq/index.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
+import { html, render } from "lit-html";
+import { compatibilityChart } from "../../components/compatibilityChart";
```
Comment (kritzcreek) [inline src/frontend/src/flows/faq/index.ts:2]: unused

File: src/frontend/src/flows/confirmRedirect.ts [hunk 9]
```diff
@@ -55,10 +56,9 @@ const pageContent = (hostName: string, principal: string) => html`
     <button id="confirmRedirect" class="primary">Proceed</button>
     <button id="cancelRedirect">Cancel</button>
     <div id="confirmRedirectPrincipal" class="highlightBox">
-      <span class="tooltip">
-        <span class="tooltiptext">${textTooltip}</span>
```
Comment (kritzcreek) [inline src/frontend/src/flows/confirmRedirect.ts:59]: `textTooltip` is unused now and should be deleted.

**Review by kritzcreek — CHANGES_REQUESTED** [review]
Looks great, this will be a big help. Got a couple of small-ish nits

File: src/frontend/src/flows/faq/index.ts (hunk 3 above)
Comment (nmattia) [reply src/frontend/src/flows/faq/index.ts:35]: Agreed! But I've just copied this from the about page. I suggest we change them both together separately.

File: src/frontend/src/flows/faq/index.ts (hunk 2 above)
Comment (nmattia) [reply src/frontend/src/flows/faq/index.ts:42]: I've just looked at a few websites (Google, Twitter, Swiss Gov, Swiss railways, my bank, basically everything I could think of) and they all have "FAQ". Does that work for you?

File: src/frontend/src/flows/faq/index.ts (hunk 5 above)
Comment (nmattia) [reply src/frontend/src/flows/faq/index.ts]: Done!

File: src/frontend/src/flows/faq/index.ts (hunk 4 above)
Comment (nmattia) [reply src/frontend/src/flows/faq/index.ts:36]: oh yeah

Comment (nmattia) [reply src/frontend/src/flows/faq/index.ts]: Done!

File: src/frontend/src/styles/main.css (hunk 7 above)
Comment (nmattia) [reply src/frontend/src/styles/main.css]: I've just pushed a commit un-nesting the selectors and fixing the margin issue (I've also added some margin at the bottom, now it doesn't look like the buttons just fell to the bottom of the page with gravity). Let me know what you think!

**Review by kritzcreek — APPROVED** [review]
LGTM, thanks!

**Pull request merged**